changes/2025-01-17_key-store-admin/background.md

```diff
@@ -0,0 +1,230 @@
+
+This option is provided by default,
+if no other `Key Management` is provided.
+
+`AwsKms` represents AWS Key Management Service (KMS).
+
+The structure has two optional fields:
+
+- `grantTokenList`, a list of strings
+- `kmsClient`, a reference to an AWS SDK KMS client
+
+If no `grantTokenList` is provided,
+it defaults to an empty list.
+
+If no `kmsClient` is provided,
+the default KMS client is constructed via the AWS SDK.
+
+## Key Management Strategy
+
+`Key Management Strategy` is a union,
+such that additional options may be added at a later date.
+
+`Key Management Strategy` determines which Operations
+of a `Key Management` are used by the Client.
+
+For example,
+an MPL Consumer MAY want to avoid calling `kms:ReEncrypt`,
+and would rather use `kms:Decrypt` followed by `kms:Encrypt`
+to re-wrap a new DECRYPT_ONLY Branch Key as a ACTIVE Branch Key.
+
+At this time,
+there are only two
+`Key Management Strategy`s.
+
+### AWS KMS ReEncrypt (default)
+
+`AwsKmsReEncrypt` dictates the Key Store Operation use
+AWS KMS' ReEncrypt Operation to
+[authenticate a Key Store Item](../../framework/branch-key-store.md#authenticating-a-keystore-item)
+or re-wrap Branch Keys
+during [Wrapped Branch Key Creation](../../framework/branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsReEncrypt` is a structure that holds a `AwsKms`,
+which MAY contain the KMS Client
+or Grant Tokens the Key Store,
+if they are set,
+will use when calling KMS.
+
+`AwsKmsReEncrypt` is the default option if
+`Key Management Strategy` is marked as optional
+and the parameter is unfilled.
+
+### AWS KMS Decrypt Encrypt
+
+`AwsKmsDecryptEncrypt` dictates the Key Store Operation to use
+AWS KMS' Decrypt Operation followed by AWS KMS Encrypt Operation
+to re-wrap Branch Keys
+during [Wrapped Branch Key Creation](../../framework/branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsDecryptEncrypt` is a structure that holds two `AwsKms`,
+one designated for Decrypt,
+one designated for Encrypt.
+
+This allows MPL Consumers to configure different
+credentials or request headers for the KMS Operations.
+
+# Modified Operations from the original [Branch Key Store Specification](../../framework/branch-key-store.md#operations)
+
+## CreateKey
+
+The CreateKey caller MUST provide:
+
+- A `KmsSymmetricKeyArn`
+
+The CreateKey caller MAY provide:
+
+- A Branch Key ID
+- Encryption Context
+- A Key Management Strategy
+
+At this time, the Key Management Strategy MUST be `AwsKmsReEncrypt`.
+The behavior is identical.
+
+## Version Key
+
+The VersionKey caller MUST provide:
+
+- A `KmsSymmetricKeyArn`
+- A `Identifier`
+
+The VersionKey caller MAY provide:
+
+- A Key Management Strategy
+
+At this time, the Key Management Strategy MUST be `AwsKmsReEncrypt`.
+The behavior is identical.
+
+<!--  LocalWords:  MRK AwsKms grantTokenList kmsClient ReEncrypt  -->
+<!--  LocalWords:  AwsKmsReEncrypt keystore AwsKmsDecryptEncrypt  -->
+<!--  LocalWords:  Admin ReEncrypt  -->
```

Comment:
```suggestion
```

Reply:
Boo... Renderded Markdown will not see this,
and `ispell` is a common spell checker.